ts/components/messages/MessageListComponent.tsx

```diff
@@ -72,17 +75,30 @@ class MessageListComponent extends React.Component<Props> {
       <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
     );
 
+    const scrollToTop = () => {
+      {
+        this.flatListRef &&
+          this.flatListRef.scrollToIndex({ animated: false, index: 0 });
+      }
+    };
+
     return (
-      <FlatList
-        contentContainerStyle={styles.contentContainerStyle}
-        scrollEnabled={true}
-        data={messages}
-        extraData={{ servicesById, paymentByRptId }}
-        keyExtractor={keyExtractor}
-        renderItem={this.renderItem}
-        refreshControl={refreshControl}
-        ListEmptyComponent={ListEmptyComponent}
-      />
+      <View>
+        <NavigationEvents onWillFocus={() => scrollToTop()} />
```

Comment:
Every time the component is rendered you are creating a new function. You can just use onWillFocus={this.scrollToTop} after moving the function as suggested above.